frontend/src/components/profile/ProfileContent.js

```diff
@@ -0,0 +1,240 @@
+  position: "absolute",
+  top: -10000,
+  left: -10000,
+}
+
+const requiredRules = [{ required: true, message: 'Missing field' }]
+
+const mapStateToProps = state => ({
+  isFetching: state.users.isFetching,
+  user: state.users.itemsByID[state.auth.currentUserID],
+  error: state.users.error,
+  groups: Object.values(state.groups.itemsByID),
+});
+
+const actionCreators = {updateSelf};
+
+const ProfileContent = ({isFetching, groups, user, error, updateSelf}) => {
+  const history = useHistory();
+  const [state, setState] = useState({ message: undefined, success: undefined });
+
+
+  /*
+   * Form Data submission
+   */
+
+  const [key, setKey] = useState(1)
+  const [formData, setFormData] = useState(deserialize(user))
+  const [formErrors, setFormErrors] = useState({})
+
+  if (formData === undefined && user !== undefined && !user.isFetching) {
+    setFormData(deserialize(user))
+    setKey(key + 1)
+  }
+
+  const onValuesChange = (values) => {
+    setFormData(deserialize({ ...formData, ...values }))
+  }
+
+  const onSubmit = () => {
+    let data
+    try {
+      data = serialize(formData, user)
+    } catch(err) {
+      setState({ message: err.message, success: false })
+      return
+    }
+    setState({ message: undefined, success: undefined })
+    updateSelf(data)
+      .then(() => {
+        setFormErrors({})
+        setState({ message: 'Profile updated', success: true })
+      })
+      .catch(err => {
+        console.log(err)
+        setFormErrors(err.data || {})
+      })
+  }
+
+  /*
+   * Render
+   */
+
+  const props = name =>
+    !formErrors[name] ? { name } : {
+      name,
+      hasFeedback: true,
+      validateStatus: 'error',
+      help: formErrors[name],
+    }
+
+  const errors = []
+    .concat(formErrors?.non_field_errors)
+    .concat(formErrors?.detail)
+    .concat(error?.message)
+    .concat(state.success === false ? state.message : [])
+    .filter(Boolean)
+
+  return (
+    <>
+      <AppPageHeader
+        title='Profile'
+        onBack={() => history.goBack()}
+      />
+      <PageContent>
+        <div>
+          <Title level={2}><UserOutlined /> {user?.username}</Title>
+          <Form
+            key={key}
+            labelCol={{ span: 4 }}
+            wrapperCol={{ span: 12 }}
+            layout="horizontal"
+            autoComplete="off"
+            initialValues={formData}
+            onValuesChange={onValuesChange}
+            onFinish={onSubmit}
+          >
+            {/* Google Chrome is very dumb and tries to autocomplete things
+              even when we explicitely set autocomplete="new-password". Thus
+              we create fake fields, so our real fields stay pristine.
+              The tabIndex & placeholder values are filled to make it keyboard
+              inaccessible. */}
+            <input
+              type="password"
+              autoComplete="password"
+              placeholder="Fake password field"
+              style={hiddenField}
```

Comment:
I am curious about this approach... Is it necessary to the app? What caught my attention the most is the in-page css '  top: -10000, left: -10000,'

Reply:
The comment explains why it's necessary.

For the inline-css, it's the way the rest of the app is styled, that I dislike but it's done for consistencty. The only places where stylesheets are used is where inline CSS cannot be used, either because the `style` attribute isn't supported or because inline styles don't allow for the required CSS features (eg, targetting a subchild when a parent has a certain calss, `.sider.expanded .title { ... }`).